### PR TITLE
Fix #107: Write format/cleanup profiles to XML files within project

### LINKED ISSUE #107: NoClassDefFoundError on generation of Eclipse formatter settings from checkstyle.xml
Eclipse 2018-09 (4.9.0) Java EE Edition
CheckStyle 8.12 plugin

Trying to generate my Eclipse formatter settings from my checkstyle.xml is generating this error.

```
!ENTRY org.eclipse.core.jobs 4 2 2018-10-22 08:39:30.452
!MESSAGE An internal error occurred during: "transformCheckstyle".
!STACK 0
java.lang.NoClassDefFoundError: org/eclipse/jdt/internal/ui/preferences/PreferencesAccess
	at net.sf.eclipsecs.core.transformer.FormatterConfigWriter.writeCleanupSettings(FormatterConfigWriter.java:95)
	at net.sf.eclipsecs.core.transformer.FormatterConfigWriter.writeSettings(FormatterConfigWriter.java:89)
	at net.sf.eclipsecs.core.transformer.FormatterConfigWriter.<init>(FormatterConfigWriter.java:81)
	at net.sf.eclipsecs.core.transformer.CheckstyleTransformer.transformRules(CheckstyleTransformer.java:124)
	at net.sf.eclipsecs.core.jobs.TransformCheckstyleRulesJob.runInWorkspace(TransformCheckstyleRulesJob.java:117)
	at org.eclipse.core.internal.resources.InternalWorkspaceJob.run(InternalWorkspaceJob.java:42)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:63)
```

### PR DESCRIPTION
**Pull request by karypid**
Instead of using the broken internal APIs, simply write directly into XML files that the user can then import.

### CODE REVIEW

File: net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java [hunk 1]
```diff
@@ -152,60 +96,18 @@ private void writeCleanupSettings(final Map<String, String> settings) {
...
-      CustomProfile myProfile = (CustomProfile) manager
-              .getProfile(ProfileManager.ID_PREFIX + mNewProfileName);
-
-      if (myProfile == null) {
-        // take current settings and create new profile
-        Profile current = manager.getSelected();
-        myProfile = new CustomProfile(mNewProfileName, current.getSettings(),
-                versioner.getCurrentVersion(), versioner.getProfileKind());
-        manager.addProfile(myProfile);
+      IFile settingsFile = mProject.getFile("checkstype-formatter-settings.xml");
```
Comment (Calixte) [inline net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java:105]: There's a typo "checkstype" vs "checkstyle".

File: net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java [hunk 2]
```diff
@@ -91,59 +79,15 @@ private void writeSettings() {
...
-      CustomProfile myProfile = (CustomProfile) manager
-              .getProfile(ProfileManager.ID_PREFIX + mNewProfileName);
-
-      if (myProfile == null) {
-        // take current settings and create new profile
-        Profile current = manager.getSelected();
-        myProfile = new CustomProfile(mNewProfileName, current.getSettings(),
-                versioner.getCurrentVersion(), versioner.getProfileKind());
-        manager.addProfile(myProfile);
+      IFile settingsFile = mProject.getFile("checkstype-cleanup-settings.xml");
```
Comment (Calixte) [inline net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java:85]: typo "checkstype"

**Comment by romani** [comment]
@karypid , please continue

**Comment by romani** [comment]
@karypid , please squash commits.
please reply "done"  to each review item.

File: net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java [hunk 3]
```diff
@@ -152,60 +96,18 @@ private void writeCleanupSettings(final Map<String, String> settings) {
    *
    * @param settings
    *          All the settings.
+   * @throws CoreException
```
[the remark this answers is not in the document]
Comment (romani) [reply net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java]: done.

File: net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java (hunk 1 above)
Comment (romani) [reply net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java]: becomes `"-cs-formatter.xml"`

File: net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java (hunk 2 above)
Comment (romani) [reply net.sf.eclipsecs.core/src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java]: becomes `"-cs-cleanup.xml"`

**Comment by romani** [comment]
I adjusted checkstyle config  at https://github.com/checkstyle/eclipse-cs/pull/261

**Comment by romani** [comment]
one more pack of violations:
```
[INFO] m--- maven-checkstyle-plugin:3.1.1:check (checkstyle-check) @ net.sf.eclipsecs.core ---

[INFO] There are 6 errors reported by Checkstyle 8.35 with config/checkstyle_checks.xml ruleset.

[ERROR] src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java:[46,3] (javadoc) JavadocVariable: Missing a Javadoc comment.

[ERROR] src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java:[46,24] (naming) AbbreviationAsWordInName: Abbreviation in name 'CS_GENETATED' must contain no more than '1' consecutive capital letters.

[ERROR] src/net/sf/eclipsecs/core/transformer/FormatterConfigWriter.java:[46,24] (naming) MemberName: Name 'CS_GENETATED' must match pattern '^(id)|([a-z][a-z0-9][a-zA-Z0-9]+)$'.

[ERROR] src/net/sf/eclipsecs/core/transformer/XmlProfileWriter.java:[108,9] (blocks) RightCurly: '}' at column 9 should be alone on a line.

[ERROR] src/net/sf/eclipsecs/core/transformer/XmlProfileWriter.java:[114,19] (coding) FinalLocalVariable: Variable 'transformer' should be declared final.

[ERROR] src/net/sf/eclipsecs/core/transformer/XmlProfileWriter.java:[119,66] (naming) CatchParameterName: Name 'e' must match pattern '^(ex|[a-z][a-z][a-zA-Z]+)$'.

```

**Comment by romani** [comment]
@Calixte , please review

**Comment by romani** [comment]
@melloware , did you tested it on your local ?
can you share some prove that it works ?

**Comment by melloware** [comment]
Sorry did not test it just did a manual eyeball code review.

**Comment by Calixte** [comment]
- fixed a typo
- fixed a bug where it would not be able to update the file
- removed unnecessary try-with-resources around byte array streams
- tested

Screenshot from 2020-09-01 08-30-36
Screenshot from 2020-09-01 08-32-42

**Pull request merged**